model/dao/TaskDAO/PostgreSQLTaskDAO.php

```diff
@@ -703,11 +702,27 @@ public function partialUpdate(DirtyTaskVO $taskVO) {
         $sql = $sql . ", updated_at=now() WHERE id=".$taskVO->getId();
 
             $res = pg_query($this->connect, $sql);
-        if ($res == NULL) throw new SQLQueryErrorException(pg_last_error());
-            $affectedRows = pg_affected_rows($res);
+        if ($res == NULL) {
+            $errorMessage = pg_last_error();
+            $resultMessage = "Task update failed:\n";
+            if(strpos($errorMessage, "end_after_init_task")) {
+                $resultMessage .= "Start time later than end time.";
+            }
+            else {
+                $resultMessage .= $errorMessage;
+            }
+            $result->setMessage($resultMessage);
+            $result->setIsSuccessful(false);
+            $result->setResponseCode(500);
```

Comment:
I know we've been using 500 everywhere but I wonder if it shouldn't be 400 in this case, what do you think? Here and in the other places where the request fails because the user sent a bad input.